### PR TITLE
[CI] Add test loop pipeline

### PR DESCRIPTION
**Pull request by v1v**
### What

New Multibranch Pipeline to support the requirement of running the UTs/ITs for a while 

- On a PR basis with the comment that contains `Jenkins run the loop tests`
- On a merge to master

_NOTE_: the GH comment was chosen in a way it does not overlap with the existing one in the main pipeline `Jenkins run the tests`

### Why

to detect any kind of flakiness 

### Issues

Closes https://github.com/elastic/apm-agent-php/issues/243

### Tasks

- Support nightly builds in a follow up
- This PR will require a follow-up

### Tests

- Run locally

```bash
$ make -f .ci/Makefile loop
```

### CODE REVIEW

File: .ci/loop.groovy [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult(prComment: true)
```
Comment (v1v) [inline .ci/loop.groovy:89]:
```suggestion
      ## PR comment is not needed with this pipeline
      notifyBuildResult(prComment: false)
```

**Review by kuisathaverat — APPROVED** [review]

**Review by SergeyKleyman — APPROVED** [review]

File: .ci/loop.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+LOOPS=${1:-50}
+DOCKERFILE=${2:-Dockerfile}
+PHP_VERSION=${3:-7.2}
+for (( c=1; c<=LOOPS; c++ ))
+do  
+    echo "Loop $c"
```
Comment (SergeyKleyman) [inline .ci/loop.sh:8]: Where can I see it being printed?

**Comment by SergeyKleyman** [comment]
Maybe we can add `Jenkins` the keyword comment? So it's something like `Jenkins run tests in loop` - this way the purpose of the comment is quite clear.

File: .ci/loop.sh (hunk 2 above)
Comment (v1v) [reply .ci/loop.sh:8]: No yet, this particular PR contains everything:
- Jenkins Job definition
- Jenkins Pipeline
- Scripts to run

Jenkins Job Definition (everything in the `.ci/jobs` folder) need to merged onto the master branch. Then the new multibranch pipeline for the loops will be created.

In other words, a kind of `chicken-egg` paradox.

**Review by cachedout — APPROVED** [review]

**Pull request merged**